packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -1271,6 +1273,47 @@ export default class FulfillmentModuleService<
     return Array.isArray(result) ? result[0] : result
   }
 
+  async retrieveFulfillmentOptions(
+    providerId: string
+  ): Promise<Record<string, any>[]> {
+    return await this.fulfillmentProviderService_.getFulfillmentOptions(
+      providerId
+    )
+  }
+
+  async validateFulfillmentOption(
+    providerId: string,
+    data: Record<string, unknown>
+  ): Promise<boolean> {
+    return await this.fulfillmentProviderService_.validateOption(
+      providerId,
+      data
+    )
+  }
+
+  @InjectManager('baseRepository_')
+  async isShippingOptionValidForContext(
+    shippingOptionId: string,
+    context: Record<string, unknown> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ) {
```

Comment:
question: Would it make sense to have: `listShippingOptionsForContext` and have the same behavior that `listShippingOptions` had before?

Reply:
Or `listValidShippingOptions`

Reply:
I prefer the first one, i can do that

Reply:
I am wondering if it makes sense to do that in truth. Why two methods one to list and one to list plus filter on context when we had one method to do this and if you dont pass the context it just performs a classic list. What do you prefer?